From 993882e7b82676c84642403be8efa1c653209204 Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Thu, 22 Aug 2024 22:50:05 +0800
Subject: [PATCH 1010/1010] ARM: dts: aspeed: Harma: enable uart dma mode

enable uart dma mode in uart4 and uart6.

BaseOn: https://lore.kernel.org/openbmc/20230320081133.23655-1-chiawei_wang@aspeedtech.com/
The DMA UART driver has not yet been merged upstream, so the patch is currently only in the meta layer.

---
 .../boot/dts/aspeed/aspeed-bmc-facebook-harma.dts    | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-harma.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-harma.dts
index dee34816aa53..b177304912fa 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-harma.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-harma.dts
@@ -86,21 +86,30 @@ tpmdev@0 {
 	};
 };
 
+&udma {
+	status = "okay";
+};
+
 // HOST BIOS Debug
 &uart1 {
+	compatible = "aspeed,ast2600-uart";
 	status = "okay";
 };
 
 // SOL Host Console
 &uart2 {
+	compatible = "aspeed,ast2600-uart";
 	status = "okay";
 	pinctrl-0 = <>;
 };
 
 // SOL BMC Console
 &uart4 {
+	compatible = "aspeed,ast2600-uart";
 	status = "okay";
 	pinctrl-0 = <>;
+	dmas = <&udma 6>, <&udma 7>;
+	dma-names = "tx", "rx";
 };
 
 // BMC Debug Console
@@ -110,7 +119,10 @@ &uart5 {
 
 // MTIA
 &uart6 {
+	compatible = "aspeed,ast2600-uart";
 	status = "okay";
+	dmas = <&udma 8>, <&udma 9>;
+	dma-names = "tx", "rx";
 };
 
 &uart_routing {
-- 
2.25.1

